### ISSUE TITLE
Cannot use mock due to private type

### ISSUE THREAD

**Issue by wackazong** [audunhalland/unimock#58]
When trying to use a mock created with 

    #[entrait::entrait(pub MyTrait, mock_api=MyTraitMock)]

for a generic function that is from another file, I could import the mock with version 0.5.0 and use it. Now, when I do that, I get a compile error

    type `my::module::path::_::__GenericMyTraitMock<T>` is private

In the macro expansion, I can see that this struct is created in an anonymous constant block and it is not defined as pub. 

I would expect the code to compile as with 0.5.0.

**Comment by audunhalland** [comment]
That was never intended to be public, as should be obvious given the `__` prefix.

**Comment by wackazong** [comment]
Yes, sure. But it is a required dependency if you want to use a Mock generated by entrait in another file. The private struct is referenced and cannot be accessed. I am not using the private strict directly. I just wanted to point you to the place where the error comes from.

**Comment by wackazong** [comment]
Here you go. 

https://github.com/wackazong/entrait-private-type

Using `cargo test` produces the following output for me:

```console
   Compiling entrait-private-type v0.1.0 (/Users/alexander/Development/Repos/ibexus/entrait-private-type)
error: type `__GenericAnotherTraitMock<T>` is private
  --> src/one_trait.rs:19:17
   |
19 |         let _ = Unimock::new(AnotherTraitMock
   |                 ^^^^^^^^^^^^ private type

error: type `__GenericAnotherTraitMock<T>` is private
  --> src/one_trait.rs:19:30
   |
19 |           let _ = Unimock::new(AnotherTraitMock
   |  ______________________________^
20 | |             .with_types::<ConcreteThing>()
21 | |             .some_call(matching!(_))
22 | |             .answers(&|_, _| ()),);
   | |________________________________^ private type

error: could not compile `entrait-private-type` (bin "entrait-private-type" test) due to 2 previous errors
```

**Comment by wackazong** [comment]
Putting a `pub(crate)` in front of the struct definition solves it for me. But how do I figure out where to patch your code? This is above me, sorry.

**Comment by audunhalland** [comment]
released 0.6.7
